src/fia/fia/factor_instance_accumulator.rs

```diff
@@ -0,0 +1,119 @@
+use super::*;
+use crate::prelude::*;
+
+/// === FIA ===
+
+pub struct FactorInstanceAccumulator<ID, Path, Product>
+where
+    ID: Hash,
+    Path: HasDerivationPath,
+    Product: HasHDPublicKey,
+{
+    state: RefCell<FiaState<ID, Path, Product>>,
+    dependencies: FiaDependencies<ID, Path, Product>,
+}
+
+/// ===== Public =====
+impl<ID, Path, Product> FactorInstanceAccumulator<ID, Path, Product>
+where
+    ID: Hash,
+    Path: HasDerivationPath,
+    Product: HasHDPublicKey,
+{
+    type State = FiaState<ID, Path, Product>;
+    type Dependencies = FiaDependencies<ID, Path, Product>;
+
+    type Driver = dyn UseFactorSourceDriver<ID, Path, Product>;
+    type BoxedDriver = Box<Self::Driver>;
+    type DriverRequest = BatchUseFactorSourceRequest<ID, Path>;
+    type DriverResponse = BatchUseFactorSourceResponse<ID, Product>;
+
+    pub fn new(
+        supports_skipping_of_factor_sources: bool,
+        request: BatchUseFactorSourceRequest<ID, Path>,
+        all_factor_sources_in_profile: impl IntoIterator<Item = FactorSource>,
+        all_drivers: impl IntoIterator<Item = Box<dyn UseFactorSourceDriver<ID, Path, Product>>>,
+    ) -> Result<Self> {
+        let factor_sources = Self::factor_sources_to_use(&request, all_factor_sources_in_profile)?;
+        let drivers = Self::drivers_to_use(&factor_sources, all_drivers)?;
+
+        let dependencies = Self::Dependencies::new(drivers);
+        let state = FiaState::new(supports_skipping_of_factor_sources, factor_sources);
+
+        Ok(Self {
+            state: RefCell::new(state),
+            dependencies,
+        })
+    }
+
+    pub async fn accumulate(&self) -> Result<FiaOutput<ID, Product>> {
```

Comment:
the single public method of `FIA`, which starts the process of accumulating FactorInstances from all FactorSources. 

Let me know if you have suggestions on improvemnt, but I think this is as clean as it gets?